include/sketch.h

```diff
@@ -13,10 +19,12 @@ using namespace std;
 class Sketch {
   const long seed;
   const uint64_t n;
-  std::vector<Bucket> buckets;
-  const long long int random_prime;
+  std::vector<Bucket_Boruvka> buckets;
+  const uint128_t large_prime;
   bool already_quered = false;
 
+  FRIEND_TEST(SketchTestSuite, TestExceptions);
```

Comment:
I'm still new to gtest, so I assume this is a gtest construct that links this code to the testing suite?

Reply:
It makes a test a friend of a class. To generate a vector where all buckets fail, we need to know which indices each bucket contains, which is private.